modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormViewFormInstanceRecordsDisplayContext.java

```diff
@@ -103,8 +103,19 @@ public DDMFormViewFormInstanceRecordsDisplayContext(
 		PortletDisplay portletDisplay = themeDisplay.getPortletDisplay();
 
 		portletDisplay.setShowBackIcon(true);
-		portletDisplay.setURLBack(
-			ParamUtil.getString(_renderRequest, "redirect"));
+
+		String redirect = ParamUtil.getString(_renderRequest, "redirect");
+
+		if (redirect.isEmpty()) {
+			DDMFormAdminDisplayContext ddmFormAdminDisplayContext =
+				(DDMFormAdminDisplayContext)_renderRequest.getAttribute(
+					WebKeys.PORTLET_DISPLAY_CONTEXT);
+
+			redirect = String.valueOf(
+				ddmFormAdminDisplayContext.getPortletURL());
```

Comment:
Hey @linolaoj and @eloisa-amorim,

Apparently, the `PortletURL` class already has a method (`toString`) to convert a Portlet URL to a String and, from what I could see, this method goes one step ahead when compared to the method `String.valueOf`. Besides, it also looks to be a common pattern on portal's code. In light of this, can any of you please confirm this information and if it's the case please use the method `toString` instead?

Thanks.

Reply:
Hi @natocesarrego , we were using the `toString`, but the `sourceFormat` task suggested using `String.valueOf`. That is the only reason why we are using `String.valueOf`. We can change back and try to run the SF again.

Reply:
Not necessary, @linolaoj. :slightly_smiling_face: Let's follow SF suggestion. Thanks for your clarification. :wink: